pkg/terra/contract_reader.go

```diff
@@ -67,66 +67,153 @@ func (r *OCR2Reader) LatestConfig(ctx context.Context, changedInBlock uint64) (t
 
 	for _, event := range res.TxResponses[0].Logs[0].Events {
 		if event.Type == "wasm-set_config" {
-			return parseAttributes(event.Attributes)
+			cc, unknown, err := parseAttributes(event.Attributes)
+			if len(unknown) > 0 {
+				r.lggr.Warnf("wasm-set_config event contained unrecognized attributes: %v", unknown)
+			}
+			return cc, err
 		}
 	}
 	return types.ContractConfig{}, fmt.Errorf("No set_config event found for tx %s", res.TxResponses[0].TxHash)
 }
 
-func parseAttributes(attrs []cosmosSDK.Attribute) (output types.ContractConfig, err error) {
+// parseAttributes returns a ContractConfig parsed from attrs.
+// An error will be returned if any of the 8 required attributes are not present, or if any duplicates are found for
+// unique attributes.
+// unknownKeys contains counts of any unrecognized keys, which are otherwise ignored.
+func parseAttributes(attrs []cosmosSDK.Attribute) (output types.ContractConfig, unknownKeys map[string]int, err error) {
+	const uniqueKeys = 8
+	known := make(map[string]struct{}, uniqueKeys)
+	first := func(key string) bool {
+		_, ok := known[key]
+		if ok {
+			return false
+		}
+		known[key] = struct{}{}
+		return true
+	}
 	for _, attr := range attrs {
 		key, value := attr.Key, attr.Value
 		switch key {
 		case "latest_config_digest":
+			if !first(key) {
+				err = ErrAttrDupe(key)
+				return
+			}
```

Comment:
Consider extracting this outside the `switch` statement, and have it declared just once under `for..range` instead of once per `case`.